src/lib/storage/dictionary_compression.cpp

```diff
@@ -99,7 +101,14 @@ class ColumnCompressor : public ColumnCompressorBase {
       }
     }
 
-    return std::make_shared<DictionaryColumn<T>>(std::move(dictionary), attribute_vector);
+    // only create statistics when the compressed dictionary is not empty
+    auto stats = dictionary.empty()
+                     ? std::shared_ptr<BaseChunkColumnStatistics>()
+                     : std::dynamic_pointer_cast<BaseChunkColumnStatistics>(
+                           std::make_shared<ChunkColumnStatistics<T>>(*dictionary.begin(), *(dictionary.end() - 1)));
```

Comment:
`dictionary.front()`, `dictionary.back()` should do the trick here as well. Also, a comment stating that those are min and max value would help i think